### ISSUE TITLE
ellipse.fit_image() doesn't work

### ISSUE THREAD

**Issue by cybertroner** [astropy/photutils#1174]
Hello fellow astronomers! I need to perform isophote analyzis for giant ellipticals in Illustris simulation, but when I try to use fit_image, it reports about division on zero somewhere in the code. When running photutils.test() it writes many 'skipped' marks in isophote package and only 'skipped' against isophote\tests\test_integrator.py. Here is the code example:

> import h5py
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.image as mpimg
import photutils
from PIL import Image
img = Image.open('vis_TNG300_1_subhalo_99_2_stars.jpg').convert('L')
from numpy import asarray
numpydata = asarray(img)
from photutils.isophote import EllipseGeometry
from photutils import EllipticalAperture
aper = EllipticalAperture((geometry.x0, geometry.y0), geometry.sma, geometry.sma*(1 - geometry.eps), geometry.pa)
plt.imshow(numpydata, origin='lower')
aper.plot(color='white')
![image](https://user-images.githubusercontent.com/80129799/110353253-e51f5000-8047-11eb-9c6b-e8ba85ee041c.png)
from photutils.isophote import Ellipse
ellipse = Ellipse(numpydata, geometry)
isolist = ellipse.fit_image()

And then goes the output

> C:\...\lib\site-packages\photutils\isophote\sample.py:335: RuntimeWarning: divide by zero encountered in double_scalars
  self.gradient_relative_error = gradient_error / np.abs(gradient)
C:\Users\cyber\.conda\envs\flame\lib\site-packages\photutils\isophote\fitter.py:317: RuntimeWarning: divide by zero encountered in double_scalars
  aux = -harmonic * (1. - sample.geometry.eps) / sample.gradient

 OverflowError                             Traceback (most recent call last)
<ipython-input-3-89fffecacc90> in <module>
----> 1 isolist = ellipse.fit_image()

~\.conda\envs\flame\lib\site-packages\photutils\isophote\ellipse.py in fit_image(self, sma0, minsma, maxsma, step, conver, minit, maxit, fflag, maxgerr, sclip, nclip, integrmode, linear, maxrit, fix_center, fix_pa, fix_eps)
    453         # now, go from initial sma inwards towards center.
    454         while True:
--> 455             isophote = self.fit_isophote(sma, step, conver, minit, maxit,
    456                                          fflag, maxgerr, sclip, nclip,
    457                                          integrmode, linear, maxrit,

~\.conda\envs\flame\lib\site-packages\photutils\isophote\ellipse.py in fit_isophote(self, sma, step, conver, minit, maxit, fflag, maxgerr, sclip, nclip, integrmode, linear, maxrit, noniterate, going_inwards, isophote_list)
    609                                            sclip, nclip, integrmode)
    610         else:
--> 611             isophote = self._iterative(sma, step, linear, geometry, sclip,
    612                                        nclip, integrmode, conver, minit,
    613                                        maxit, fflag, maxgerr, going_inwards)

~\.conda\envs\flame\lib\site-packages\photutils\isophote\ellipse.py in _iterative(self, sma, step, linear, geometry, sclip, nclip, integrmode, conver, minit, maxit, fflag, maxgerr, going_inwards)
    633             fitter = CentralEllipseFitter(sample)
    634 
--> 635         isophote = fitter.fit(conver, minit, maxit, fflag, maxgerr,
    636                               going_inwards)
    637 

~\.conda\envs\flame\lib\site-packages\photutils\isophote\fitter.py in fit(self, conver, minit, maxit, fflag, maxgerr, going_inwards)
    222             # the EllipseSample constructor.
    223             sample = corrector.correct(sample, largest_harmonic)
--> 224             sample.update(fixed_parameters)
    225 
    226             # see if any abnormal (or unusual) conditions warrant

~\.conda\envs\flame\lib\site-packages\photutils\isophote\sample.py in update(self, fixed_parameters)
    297 
    298         # Update the mean value first, using extraction from main sample.
--> 299         s = self.extract()
    300         self.mean = np.mean(s[2])
    301 

~\.conda\envs\flame\lib\site-packages\photutils\isophote\sample.py in extract(self)
    150             return self.values
    151         else:
--> 152             s = self._extract()
    153             self.values = s
    154             return s

~\.conda\envs\flame\lib\site-packages\photutils\isophote\sample.py in _extract(self, phi_min)
    213             # do the integration at phi-radius position, and append
    214             # results to the angles, radii, and intensities lists.
--> 215             integrator.integrate(radius, phi)
    216 
    217             # store sector area locally

~\.conda\envs\flame\lib\site-packages\photutils\isophote\integrator.py in integrate(self, radius, phi)
    176         x_ = radius * math.cos(phi + self._geometry.pa) + self._geometry.x0
    177         y_ = radius * math.sin(phi + self._geometry.pa) + self._geometry.y0
--> 178         i = int(x_)
    179         j = int(y_)
    180         fx = x_ - i

OverflowError: cannot convert float infinity to integer

**Comment by larrybradley** [comment]
@ibusko Can you take a look?

@cybertroner The skipped tests rely on remote data. If you want to run the full test suite, run `photutils.test(remote_data=True)`.

**Comment by ibusko** [comment]
@cybertroner there is a bug in the code that acts when the gradient is exactly zero. That seems to be the case with your combination of image and starting geometry parameters. I am putting in place a fix; it should be available with the next release of 'photutils'. 

In the meantime, you can try to use slightly different values for your starting geometry parameters; this may avoid tripping the bug. 

Thanks for reporting the bug!

**Comment by ibusko** [comment]
It looks like all these errors are coming from the network: `urllib.error.HTTPError: HTTP Error 404: Not Found`

**Comment by cybertroner** [comment]
Hello! That's me again. I've changed all the files according to your addition, reinstalled photutils and tried to analyze the same galaxy. Initially I received some new errors, but after some playing with fitting parameters I finally got the isophotes! It works! Here are the galaxy, the model from isophotes' parameters and residuals. I took quite large minsma (~10) since it didn't work for a smaller parameter.
![image](https://user-images.githubusercontent.com/80129799/110999366-4f5e2a80-8391-11eb-892d-9fe75607d666.png)
It appeared that the galaxy is likely lenticular. :) 
Thank you so much! I hope it'll works for other galaxies and eventually I and my advisor will make a good publication. We'll definitely cite Photutils!
